File: dynamic-thread-pool-spring-boot-starter/src/main/java/com/zeddic/dynamic/thread/pool/domain/model/entity/ThreadPoolConfigEntity.java
package com.zeddic.dynamic.thread.pool.domain.model.entity;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * @author: zeddic
 * @description: 线程池配置的实体对象
 * @date: 2024/7/10 下午4:38
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ThreadPoolConfigEntity {
    /**
     * 应用名称
     */
    private String appName;

    /**
     * 线程池名称
     */
    private String threadPoolName;

    /**
     * 核心线程数
     */
    private int corePoolSize;

    /**
     * 最大线程数
     */
    private int maximumPoolSize;

    /**
     * 当前活跃线程数
     */
    private int activeCount;

    /**
     * 当前池中线程数
     */
    private int poolSize;

    /**
     * 队列类型
     */
    private String queueType;

    /**
     * 当前队列任务数
     */
    private int queueSize;

    /**
     * 队列剩余任务数
     */
    private int remainingCapacity;

}

File: dynamic-thread-pool-front-end/src/type/ThreadPoolConfigEntity.ts
/**
 * @author:     zeddic
 * @description:  线程池配置实体类
 * @date:    2024/7/12 上午10:29
 */

export interface ThreadPoolConfigEntity {
    appName: string;
    threadPoolName: string;
    corePoolSize: number;
    maximumPoolSize: number;
    activeCount: number;
    poolSize: number;
    queueType: string;
    queueSize: number;
    remainingCapacity: number;
}